### PR TITLE
fix: get production phase data using details parameter

### PR DESCRIPTION
**Pull request by catsmanac**
[Home Assistant Issue 114862](https://github.com/home-assistant/core/issues/114862) reports that production phase entities are missing while net energy phase data is reported. Turns out that with fw 7 and up, the `/production` endpoint only reports phase details for production and consumption when using `/production.json?details=1`.

This PR implements such a change:

- A new updater `EnvoyProductionJsonDetailsUpdater(EnvoyProductionUpdater)` is added
  - Inserted in the updater list between `EnvoyMetersUpdater` and `EnvoyProductionUpdater`
  - Uses endpoint `/production.json?details=1`
  - It will only be used as the production updater if the EnvoyMetersUpdater found 2 or 3 phases
    - no probe done for the endpoint if not 2 or 3 phases reported 

This will result in:

- The `raw` entry will be `raw["/production.json?details=1"]` when using this endpoint
- All current split and multi phase systems will have phase entities added for consumption and/or production
  - requires HA [PR115577](https://github.com/home-assistant/core/pull/115577) implemented with or before this one bumps in HA, to have these disabled by default.

Other related changes:
- updated tests and snapshot
- fixtures `7.3.466_with_cts_3phase` and  `7.6.175_with_cts_3phase` are updated to not contain phase details in `production` file and have phase details in `production.json` file.
- moved actual phase count detection from production.py `update` to `probe`.

New fixtures added in the future will need to account for this behavior and implement this in the production and production.json fixture files.

### CODE REVIEW

File: src/pyenphase/updaters/production.py [hunk 1]
```diff
@@ -82,6 +96,8 @@ async def probe(
                     if type_["type"] == "eim" and type_["activeCount"]:
                         self._supported_features |= SupportedFeatures.METERING
                         self._supported_features |= SupportedFeatures.PRODUCTION
+                        if lines := type_.get("lines"):
```
Comment (bdraco) [inline src/pyenphase/updaters/production.py:99]: Why does this have a type ignore?

File: src/pyenphase/updaters/production.py [hunk 2]
```diff
@@ -107,9 +123,18 @@ async def probe(
                 if not discovered_net_consumption and meter_type == "net-consumption":
                     self._supported_features |= SupportedFeatures.NET_CONSUMPTION
 
+                if lines := meter.get("lines"):
+                    active_phase_count = len(lines)  # type: ignore[arg-type]
```
Comment (bdraco) [inline src/pyenphase/updaters/production.py:127]: Same

File: src/pyenphase/updaters/production.py (hunk 1 above)
Comment (catsmanac) [reply src/pyenphase/updaters/production.py:99]: Mypy complained on it not being Sized. It all worked but was caught in commit. Tried various options, but with my limited fundamental understanding of it I ended with that for now. Lines is [{},{},{}] of {str, str | float | int}.

afbeelding

File: src/pyenphase/updaters/production.py (hunk 2 above)
Comment (catsmanac) [reply src/pyenphase/updaters/production.py:127]: see above

File: src/pyenphase/updaters/production.py (hunk 1 above)
Comment (catsmanac) [reply src/pyenphase/updaters/production.py:99]: Added Any type to definition, and assert for list with #  nosec when type checking. That passed and looks like nosec is used in more spots.

Comment (bdraco) [reply src/pyenphase/updaters/production.py:99]: If you add a type for `lines` you shouldn't need the ignore.

`Any` effectively turns off the type checking

**Comment by cgarwood** [comment]
I haven't kept up on the production and metering endpoints, but what happens if we just include ?details=1 on the regular production.json updater? May help simplify things by not needing a new updater class

**Comment by catsmanac** [comment]
> I haven't kept up on the production and metering endpoints, but what happens if we just include ?details=1 on the regular production.json updater? May help simplify things by not needing a new updater class

Didn't dare to take that step. The existing order was
```py
    EnvoyMetersUpdater,
    EnvoyProductionUpdater,
    EnvoyProductionJsonUpdater,
```
The regular production.json is a fallback for the production one in the existing order. So the order would need to switch. Couldn't fully asses impact of this for older versions. So added the new one that only takes over when the /meters reports installed CT going by the assumption that CT's primarily used with more recent firmware's. If no CT is installed, the prior existing order remains in effect.

If you all feel this is not an issue I can change it as you are right, it would be simpler overall.

**Comment by catsmanac** [comment]
Looks like your proposal will work. Need to do some more test, but so far so good.

**Comment by catsmanac** [comment]
Reworked it and made productionjsonupdater use details=1 and moved it before productionupdater in the updaters list. 

All tests passed after replacing supportedfeatures productionupdater by productionjsonupdater  and accepting small snapshot value changes. These value changes are caused by small time differences in the production and production.json fixture files.

Looking at the test results and the fact nowhere the productionupdater is required in the tests, one would think the productionupdater for /production can be removed altogether from the updaters list.

**Comment by leah-potato** [comment]
Any luck getting this merged? I've had the same issue [as mentioned here](https://github.com/home-assistant/core/issues/114862) - can see consumption for the different phases, but not production. Would love if someone could review and get this in. Thank you!

**Review by bdraco — APPROVED** [review]
Firmware: 3.9.36 ok
Firmware: 7.3.621 ok
Firmware: 5.0.62 ok

**Pull request merged**